pkg/terraform/exec/plugins/vsphereprivate/internal/archive.go

```diff
@@ -0,0 +1,205 @@
+/*
+Copyright (c) 2014-2015 VMware, Inc. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Modified From:
+// https://github.com/vmware/govmomi/blob/e651475773eba5139c706613b3b035377d6c1c30/govc/importx/archive.go
+
+package vsphereprivate
+
+import (
+	"archive/tar"
+	"bytes"
+	"context"
+	"errors"
+	"flag"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/url"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+
+	"github.com/vmware/govmomi/ovf"
+	"github.com/vmware/govmomi/vim25"
+	"github.com/vmware/govmomi/vim25/soap"
+)
+
+// ArchiveFlag doesn't register any flags;
+// only encapsulates some common archive related functionality.
+type ArchiveFlag struct {
+	Archive
+}
+
+func newArchiveFlag(ctx context.Context) (*ArchiveFlag, context.Context) {
+	return &ArchiveFlag{}, ctx
+}
+
+// Register method
+func (f *ArchiveFlag) Register(ctx context.Context, fs *flag.FlagSet) {
+}
+
+// Process method
+func (f *ArchiveFlag) Process(ctx context.Context) error {
+	return nil
+}
+
+// ReadOvf method
+func (f *ArchiveFlag) ReadOvf(fpath string) ([]byte, error) {
+	r, _, err := f.Open(fpath)
+	if err != nil {
+		return nil, err
+	}
+	defer r.Close()
+
+	return ioutil.ReadAll(r)
```

Comment:
hmm.. this is going to load up a lot data into memory.. how big is OVF file typically :P

Reply:
@abhinavdahiya 

```
$ ls -alh ~/Downloads/rhcos-44.81.202003062006-0-vmware.x86_64.ova
-rw-rw-r--. 1 jcallen jcallen 840M Mar  6 16:59 /var/home/jcallen/Downloads/rhcos-44.81.202003062006-0-vmware.x86_64.ova
```